spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/annotation/SqsListenerAnnotationBeanPostProcessor.java

```diff
@@ -106,12 +111,23 @@ protected Collection<HandlerMethodArgumentResolver> createAdditionalArgumentReso
 
 	@Override
 	protected Collection<HandlerMethodArgumentResolver> createAdditionalArgumentResolvers(
-			MessageConverter messageConverter, ObjectMapper objectMapper) {
+			MessageConverter messageConverter, AbstractMessageConverterFactory wrapper) {
 		List<HandlerMethodArgumentResolver> argumentResolvers = new ArrayList<>(createAdditionalArgumentResolvers());
-		if (objectMapper != null) {
-			argumentResolvers.add(new NotificationMessageArgumentResolver(messageConverter, objectMapper));
-			argumentResolvers.add(new NotificationSubjectArgumentResolver(objectMapper));
-			argumentResolvers.add(new SnsNotificationArgumentResolver(messageConverter, objectMapper));
+		if (wrapper instanceof JacksonJsonMessageConverterFactory) {
```

Comment:
How about instead of having these instanceof calls, we add a `createAdditionalArgumentResolvers` method in the Factory?

This way we keep Jackson2 and Jackson3-specific code encapsulated and avoid having a CNFE from loading the specific classes.